### PR TITLE
Updates to Grades, Stats, Student Progress, and LTIUpdate Help.

### PR DESCRIPTION
**Pull request by somiaj**
Besides updating the instructor help files, this adds a help icon
  to the Problems header on the tables that show per problem grades
  to describe what the numbers mean.  The help is put there since it
  should also be visible to students when they are viewing the table.

  There was some text added to the student progress and grades pages
  pages that was essentially help that has been removed and placed
  in the instructor help file.

  Last I reduced the amount of information on the headers of the
  student progress pages to be more in line with the stats page to
  only list the set or student being viewed.

### CODE REVIEW

File: lib/WeBWorK/ContentGenerator/Grades.pm [hunk 1]
```diff
@@ -42,18 +42,7 @@ sub scoring_info ($c) {
...
-				'There is no additional grade information.  A message about additional grades can go in '
-					. '~[TMPL~]/email/[_1]. It is merged with the file ~[Scoring~]/[_2]. These files can be '
-					. 'edited using the "Email" link and the "File Manager" link in the left margin.',
-				$message_file, $merge_file
-			);
-		} else {
-			return '';
-		}
-	}
+	return if (!(-e "$ce->{courseDirs}{scoring}/$merge_file" && -e "$filePath"));
```
Comment (drgrice1) [inline lib/WeBWorK/ContentGenerator/Grades.pm]: If you are going to delete these messages that is shown when the files are not present, then you should also delete the message that is added for those with instructor permissions when the files are present.  That is also not needed if this information is added to the help file.  That is, delete lines 106-119.  Then eliminate the `$output` variable, and just return the result on line 100 (and also delete the current line 121).

Although, I am not sure you should remove these messages from the page.  It doesn't hurt to have this on the page as well as in the help.  In fact it is probably better to have it both places.

Also if these messages are deleted, this should return the empty string, not undef.  Line 58 below should also return the empty string.  These return values are used as strings.  Mojolicious templates hide the warnings, but they should still be strings.

File: lib/WeBWorK/ContentGenerator/Instructor/StudentProgress.pm [hunk 2]
```diff
@@ -177,14 +175,14 @@ sub displaySets ($c) {
...
-				incorrect_attempts => ''
+				version                    => $vNum,
+				score                      => $score,
+				total                      => $total,
+				date                       => $dateOfTest,
+				testtime                   => $testTime,
+				timeleft                   => $timeLeft,
+				problem_scores             => $problem_scores,
+				problem_incorrect_attempts => $problem_incorrect_attempts
 			};
```
Comment (drgrice1) [inline lib/WeBWorK/ContentGenerator/Instructor/StudentProgress.pm]: None of the changes to this file should be in this pull request.  This is unrelated to the help file update.  It should be in a separate pull request so that others can review the changes you are making here.

Furthermore, I completely disagree with removing the close date from the page.  Removing it from the title is fine, but removing it from the page entirely is not.

File: templates/ContentGenerator/Instructor/StudentProgress/set_progress.html.ep [hunk 3]
```diff
@@ -66,30 +66,6 @@
...
-			) =%>
-		</p>
-	% } else {
-		<p>
-			<%= maketext(
-				q{Click a student's name to see the student's homework set. Click a heading to sort the table.}
-			) =%>
-		</p>
-	% }
-	%
```
Comment (drgrice1) [inline templates/ContentGenerator/Instructor/StudentProgress/set_progress.html.ep]: I think that you should leave this here (except the part about the period indicating an unattempted problem part).  It doesn't hurt to have the help in the page as well as in the help file.  In fact it is probably better to have it both places.

File: templates/HelpFiles/Grades.html.ep [hunk 4]
```diff
@@ -16,8 +16,23 @@
...
+</p>
+<p class="m-0">
+	<%== maketext(
+		'Additional (external) grades can be shown on this page by placing them in the CSV file [_1].  The first six '
+			. 'columns must be (in order): Student ID, Username, Last Name, First Name, Section, Recitation.  The '
+			. 'remaining columns can list any external grades.  To display the grades, the CSV file is merged with '
+			. 'the "Email" message [_2], which will be rendered and displayed below the grade table.  External grades '
+			. 'can only be displayed and are not included in any totals or statistics.',
+		'<code>[Scoring]/report_grades_data.csv</code>',
+		'<code>[TMPL]/email/report_grades.msg</code>',
```
Comment (drgrice1) [inline templates/HelpFiles/Grades.html.ep]: It would be good to add something about how and where these files can be created.  The email merge message file can be created on the "Email" page, and the grades report data file can be generated on the "Scoring Tools" page.

**Review by drgrice1 — CHANGES_REQUESTED** [review]
There are some changes that are needed on this I think.  Be careful with changing other files in this effort.  Try to restrict to the HelpFiles.  Changes elsewhere can be done in other pull requests to the usual webwork branches so that others can also review them.

**Comment by somiaj** [comment]
I removed all the other non-help changes and will put them in other pull requests so they can be discussed there.

File: templates/HelpFiles/Grades.html.ep (hunk 4 above)
Comment (somiaj) [reply templates/HelpFiles/Grades.html.ep]: Added.

File: templates/ContentGenerator/Instructor/StudentProgress/set_progress.html.ep (hunk 3 above)
Comment (somiaj) [reply templates/ContentGenerator/Instructor/StudentProgress/set_progress.html.ep]: I thought the page looked nicer without it since the information if needed was in the help popup.  Though you are correct doesn't hurt anything, so I have only taken out the single line about how to read the numbers in the problems table.

File: lib/WeBWorK/ContentGenerator/Instructor/StudentProgress.pm (hunk 2 above)
Comment (somiaj) [reply lib/WeBWorK/ContentGenerator/Instructor/StudentProgress.pm]: Will put this in a new pullrequest.

**Comment by drgrice1** [comment]
Thanks for making those changes.  I just don't want to much to go into the instructor-help-update branch aside from the actual help file updates.  Things that are closely tied to those updates like the show/hide site description button removal, or the in page popovers added here are fine.  However, the other things should be done separately.

File: lib/WeBWorK/ContentGenerator/Grades.pm (hunk 1 above)
Comment (somiaj) [reply lib/WeBWorK/ContentGenerator/Grades.pm]: I have always found the message confusing and was unsure what it even meant. I also had a hard time trying to find info on how to actually add external grades and what they do when writing the help popup. Though this is another discussion, so I left it for now. I would suggest at least maybe rephrasing the messages.

File: templates/ContentGenerator/Instructor/StudentProgress/set_progress.html.ep (hunk 3 above)
Comment (drgrice1) [reply templates/ContentGenerator/Instructor/StudentProgress/set_progress.html.ep]: That may be.  Add this in another pull request for consideration.

File: lib/WeBWorK/ContentGenerator/Grades.pm (hunk 1 above)
Comment (drgrice1) [reply lib/WeBWorK/ContentGenerator/Grades.pm]: Yes.  It is confusing and certainly unclear.  It needs to be improved.
